packages/ember-core/src/components/form/search.gts

```diff
@@ -0,0 +1,428 @@
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action, get } from '@ember/object';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { restartableTask, timeout } from 'ember-concurrency';
+import { t } from 'ember-intl';
+// @ts-expect-error Ember keyboard doesn't currently ship a type for the `on-key` modifier
+// https://github.com/adopted-ember-addons/ember-keyboard/issues/464
+import onKey from 'ember-keyboard/modifiers/on-key';
+
+import BoundValue from './bound-value.ts';
+import onClickOutside from '../../modifiers/on-click-outside.ts';
+import onInsert from '../../modifiers/on-insert.ts';
+import Button from '../button.gts';
+
+import type { Optional } from '../../';
+import type IntlService from 'ember-intl/services/intl';
+
+declare type SearchOption<T> = {
+  label: string;
+  value: string | T;
+  raw: T;
+};
+
+interface SearchItemSignature<T> {
+  Args: {
+    option: SearchOption<T>;
+    index: number;
+    activeIndex?: number;
+  };
+  Element: HTMLLIElement;
+}
+
+class SearchItem<T> extends Component<SearchItemSignature<T>> {
+  get classList() {
+    const classes = ['dropdown-item'];
+
+    if (this.isActive) {
+      classes.push('active');
+    }
+
+    return classes.join(' ');
+  }
+
+  get isActive() {
+    return this.args.activeIndex === this.args.index;
+  }
+
+  <template>
+    <li
+      class={{this.classList}}
+      role="option"
+      aria-selected={{this.isActive}}
+      ...attributes
+    >{{@option.label}}</li>
+  </template>
+}
+
+export interface SearchSignature<T> {
+  Args: {
+    basic?: boolean;
+    clearable?: boolean;
+    describedBy?: string;
+    disabled?: boolean;
+    displayPath?: string;
+    format?: ((value: Optional<string>) => string) | false;
+    hideSearchIcon?: boolean;
+    id?: string;
+    isInvalid?: boolean;
+    isWarning?: boolean;
+    loading?: boolean;
+    minCharacters?: number;
+    noResultsLabel?: string;
+    placeholder?: string;
+    query: (searchString: string) => Promise<T[]>;
```

Comment:
Actions should start with `on`, e.g.n `onQuery`. Also, arguments listed in the signature should place actions at the bottom - check [`bound-value`](https://github.com/knoxville-utilities-board/nrg-ui/blob/accf736883c28f01e934bfb13b402a8467007917/packages/ember-core/src/components/form/bound-value.ts#L10-L21):

https://github.com/knoxville-utilities-board/nrg-ui/blob/accf736883c28f01e934bfb13b402a8467007917/packages/ember-core/src/components/form/bound-value.ts#L10-L21